pyracf/common/improper_userid_error.py

```diff
@@ -0,0 +1,17 @@
+"""Exception to use when attempting to set an invalid userid for pyRACF to run under."""
+
+
+class ImproperUserIDError(Exception):
+    """
+    Raised when pyRACF would attempt to establish itself to
+    run under a passed 'userid' that is not a string of 1 to 8 characters.
+    """
+
+    def __init__(self, userid: str) -> None:
+        self.message = (
+            "Cannot run under this userid."
+            + f"{userid} is not a string from 1 to 8 characters in length."
+        )
```

Comment:
```python
    """
    Raised when pyRACF would attempt to run as a userid
    that is not a string between 1 to 8 characters in length.
    """

    def __init__(self, userid: str) -> None:
        self.message = (
            "Unable to run as userid '{userid}'. Userid must "
            + f"be a string value between 1 to 8 characters in length."
        )
```

Reply:
change made.

Reply:
Can you also change `ImproperUserIdError` to `UserIdError`? I don't know if the token `Improper` is necessary. I think `UserIdError` gets the point across.

Reply:
Fair point; done!